member-service/src/main/java/com/hedvig/memberservice/aggregates/MemberAggregate.kt

```diff
@@ -0,0 +1,588 @@
+            applyChain = apply(SSNUpdatedEvent(id, ssn, SSNUpdatedEvent.Nationality.SWEDEN))
+            // -- Tracking id generation for the new member id
+            generateTrackingId()
+            applyChain = getBirthdateFromSwedishSsnAndApplyEvent(ssn, applyChain)
+            applyChain = try {
+                getPersonInformationFromBisnode(applyChain, ssn)
+            } catch (ex: RuntimeException) {
+                logger.error("Caught exception calling bisnode for personalInformation", ex)
+                applyChain!!.andThenApply {
+                    NameUpdatedEvent(
+                        id,
+                        formatName(bankIdAuthResponse.givenName),
+                        formatName(bankIdAuthResponse.surname))
+                }
+            }
+            if (status == MemberStatus.INITIATED) {
+                applyChain = applyChain!!.andThenApply { MemberStartedOnBoardingEvent(id, MemberStatus.ONBOARDING) }
+            }
+        }
+        val authenticatedEvent = MemberAuthenticatedEvent(id, bankIdAuthResponse.referenceToken)
+        if (applyChain != null) {
+            applyChain.andThenApply(Supplier<Any> { authenticatedEvent })
+        } else {
+            apply(authenticatedEvent)
+        }
+    }
+
+    private fun getBirthdateFromSwedishSsnAndApplyEvent(ssn: String?, @Nullable applyChain: ApplyMore?): ApplyMore? {
+        var applyChain = applyChain
+        val birthDate = getBirthdateFromSwedishSsn(ssn)
+        if (birthDate != null) {
+            if (applyChain != null) {
+                applyChain = applyChain.andThenApply(
+                    Supplier<Any> { BirthDateUpdatedEvent(id, birthDate) }
+                )
+                return applyChain
+            }
+            apply(BirthDateUpdatedEvent(id, birthDate))
+        }
+        return applyChain
+    }
+
+    private fun generateTrackingId() {
+        apply(TrackingIdCreatedEvent(id, uuidGenerator.generateRandom()))
+    }
+
+    private fun formatName(name: String): String {
+        val lowercase = name.toLowerCase()
+        return Character.toUpperCase(lowercase[0]).toString() + lowercase.substring(1)
+    }
+
+    @Throws(RuntimeException::class)
+    private fun getPersonInformationFromBisnode(applyChain: ApplyMore?, ssn: String): ApplyMore? {
+        var chain = applyChain
+        logger.info("Calling bisnode for person information for {}", ssn)
+        val personList = bisnodeClient.match(ssn).persons
+        val person = personList[0].person
+        if (personList.size != 1) {
+            throw RuntimeException("Could not find person at bisnode.")
+        }
+        chain = chain!!.andThenApply { NameUpdatedEvent(id, person.preferredOrFirstName, person.familyName) }
+        val pi = BisnodeInformation(ssn, person)
+        if (pi.address.isPresent) {
+            chain = chain.andThenApply(
+                Supplier<Any> { LivingAddressUpdatedEvent(id, pi.address.get()) })
```

Comment:
Not sure we need a supplier here, if not:
```suggestion
                LivingAddressUpdatedEvent(id, pi.address.get()))
```